src/ServiceControl.Monitoring/QueueLength/QueueLengthDataStore.cs

```diff
@@ -11,22 +13,57 @@ public QueueLengthDataStore(IQueueLengthCalculator calculator)
             this.calculator = calculator;
         }
 
-        public IEnumerable<KeyValuePair<string, JObject>> Current
+        public Dictionary<string, Dictionary<DateTime, double>> GetQueueLengths(DateTime now)
         {
-            get { return calculator.GetQueueLengths().Select(kvp => new KeyValuePair<string, JObject>(kvp.Key, new JObject(new JProperty("Count", kvp.Value)))); }
+            var results = new Dictionary<string, Dictionary<DateTime, double>>();
+
+            foreach (var endpointName in queueLengths.Keys)
+            {
+                ConcurrentDictionary<DateTime, double> endpointData;
+
+                if (queueLengths.TryGetValue(endpointName, out endpointData))
+                {
+                    var endpointDataSnapshot = endpointData.ToArray();
```

Comment:
Could you describe the approach applied in here? Is it "entries fresh enough to be considered + a cleanup phase"? I think I need a better understanding to yay/nay it.